### PR TITLE
feat: improve dataset command output

### PR DESCRIPTION
**Pull request by mohammad-sdsc**
# Description

Minor fixes to dataset commands output. To address long output list, this PR adds a column selection option to `renku dataset` and `renku dataset ls-files` commands where user can use to show a selected list of properties. 

Fixes https://github.com/SwissDataScienceCenter/renku-python/issues/821

### CODE REVIEW

File: renku/core/commands/format/dataset_files.py [hunk 1]
```diff
@@ -52,8 +57,33 @@ def jsonld(client, records):
     return dumps(data, indent=2)
 
 
+def _make_headers(columns):
```
Comment (Panaetius) [inline renku/core/commands/format/dataset_files.py]: This method is identical to the `_make_headers` for `DatasetFile` except for `DATASET_COLUMNS` vs `DATASET_FILES_COLUMNS`. We can have a generic `make_headers` method that takes `DATASET_COLUMNS`/`DATASET_FILES_COLUMNS` as additional input parameter to reduce code duplication.

File: renku/core/commands/format/datasets.py [hunk 2]
```diff
@@ -20,30 +20,46 @@
...
+    headers = _make_headers(columns)
+
+    # Sort based on the first requested field
+    attr = list(headers.keys())[0]
+    datasets = sorted(datasets, key=lambda d: getattr(d, attr))
+
+    return tabulate(datasets, headers=headers, disable_numparse=True)
+
+
+def _make_headers(columns):
```
Comment (Panaetius) [inline renku/core/commands/format/datasets.py]: See other comment

File: tests/cli/test_datasets.py [hunk 3]
```diff
@@ -268,19 +268,47 @@ def test_datasets_list_empty(output_format, runner, project):
...
+            'dataset', 'create', 'my-dataset', '--title', 'Long Title', '-c',
+            'John Doe <john.doe@mail.ch>'
+        ]
+    )
+    assert 0 == result.exit_code
+
+    result = runner.invoke(cli, ['dataset', '--columns', columns])
+    assert 0 == result.exit_code
+    for value in values:
+        assert value in result.output
```
Comment (Panaetius) [inline tests/cli/test_datasets.py]: This test would pass if `--columns` is ignored and all columns are output, we should also test that the excluded columns are not present.

File: renku/core/commands/format/dataset_files.py (hunk 1 above)
Comment (mohammad-sdsc) [reply renku/core/commands/format/dataset_files.py]: I've actually factored out `tabular` function which also has a similar pattern in both files.

File: tests/cli/test_datasets.py (hunk 3 above)
Comment (mohammad-sdsc) [reply tests/cli/test_datasets.py]: Good point! I've modified this test and added new ones just for that purpose.

**Pull request merged**